src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,402 @@
+                {
+                    theme = "Light";
+                }
+                else if(dictionarySource.Contains("Fluent.Dark", StringComparison.OrdinalIgnoreCase))
+                {
+                    theme = "Dark";
+                }
+                else
+                {
+                    theme = "System";
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private static Uri GetFluentThemeResourceUri(bool useLightMode)
+    {
+        string themeFileName = "Fluent." + (useLightMode ? "Light" : "Dark") + ".xaml";
+
+        if(SystemParameters.HighContrast)
+        {
+            themeFileName = "Fluent.HC.xaml";
+        }
+
+        return new Uri(fluentThemeResoruceDictionaryUri + themeFileName, UriKind.Absolute);
+    }  
+
+    private static void AddOrUpdateThemeResources(ResourceDictionary rd, Uri dictionaryUri)
+    {
+        if(rd == null) return;
+
+        ArgumentNullException.ThrowIfNull(dictionaryUri, nameof(dictionaryUri));
+
+        var newDictionary = new ResourceDictionary() { Source = dictionaryUri };
+
+        int index = FindLastFluentThemeResourceDictionaryIndex(rd);
+        
+        if (index != -1)
+        {
+            rd.MergedDictionaries.RemoveAt(index);
+            rd.MergedDictionaries.Insert(index, newDictionary);
+        }
+        else
+        {
+            rd.MergedDictionaries.Add(newDictionary);
+        }
+    }
+
+    private static int FindLastFluentThemeResourceDictionaryIndex(ResourceDictionary rd)
+    {
+        if(rd == null) return -1;
+
+        for(int i = rd.MergedDictionaries.Count - 1; i >= 0; i--)
+        {
+            if(rd.MergedDictionaries[i].Source != null)
+            {
+                if(rd.MergedDictionaries[i].Source.ToString().Contains(fluentThemeResoruceDictionaryUri))
+                {
+                    return i;
+                }
+            }
+        }
+        return -1;
+    }
+
+    private static bool IsSystemThemeLight()
+    {
+        var useLightTheme = Registry.GetValue(_regPersonalizeKeyPath,
+            "AppsUseLightTheme", null) as int?;
+
+        if (useLightTheme == null)
+        {
+            useLightTheme = Registry.GetValue(_regPersonalizeKeyPath,
+                "SystemUsesLightTheme", null) as int?;
+        }
+
+        return useLightTheme != null && useLightTheme != 0;
+    }    
+
+    #endregion
+
+    #region Private Fields
+    private static bool _deferThemeLoading = false;
+    private static readonly string fluentThemeResoruceDictionaryUri = "pack://application:,,,/PresentationFramework.Fluent;component/Themes/";
+    internal static WindowCollection _fluentEnabledWindows = new WindowCollection();
+    private static readonly string _regPersonalizeKeyPath = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
```

Comment:
just make sure this path works for both x86 and x64 apps

Reply:
Same path